File: js/바이러스(2606) copy.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input[0]);
const m = Number(input[1]);
const graph = Array.from({length: n+1}, () => []);

for (let i=2; i<m+2; i++) {
    const [x,y] = input[i].split(' ').map(Number);
    graph[x].push(y);
    graph[y].push(x);
}

const visited = Array.from({length: n+1}, () => false);

function bfs(v) {
    const queue = [];
    queue.push(v);
    visited[v] = true;

    let cnt = 0;

    while (queue.length) {
        let a = queue.shift();
        for (let i of graph[a]) {
            if (!visited[i]) {
                visited[i] = true;
                cnt++;
                queue.push(i);
            }
        }
    }
    console.log(cnt);
}

bfs(1);

File: python/바이러스(2606) copy.py
from collections import deque

n = int(input()) # 컴퓨터의 수
m = int(input()) # 네트워크 상 연결되어있는 컴퓨터 쌍 수
graph = [[] for _ in range(n+1)] 

for _ in range(m):
    x, y = tuple(map(int, input().split()))
    graph[x].append(y)
    graph[y].append(x)

def bfs(x):
    queue = deque([x])
    visited = [False] * (n+1)
    visited[x] = True
    cnt = 0

    while queue:
        y = queue.popleft()
        for k in graph[y]:
           if not visited[k]:
               visited[k] = True
               cnt += 1
               queue.append(k) 
    print(cnt)

bfs(1)
